src/lints/enum_non_exhaustive_struct_variant_field_added.ron

```diff
@@ -0,0 +1,75 @@
+    human_readable_name: "pub enum #[non_exhaustive] struct variant field added",
+    description: "A pub enum's #[non_exhaustive] struct variant has a new field.",
+    required_update: Minor,
+    lint_level: Allow,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        enum_name: name @output
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        variant {
+                            ... on StructVariant {
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+                                variant_name: name @output @tag
+
+                                field {
+                                    field_name: name @output @tag
+
+                                    span_: span @optional {
+                                        filename @output
+                                        begin_line @output
+                                        end_line @output
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            baseline {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        variant {
+                            ... on StructVariant {
+                                name @filter(op: "=", value: ["%variant_name"])
+                                attrs @filter(op: "contains", value: ["$non_exhaustive"])
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+
+                                field @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                                    name @filter(op: "=", value: ["%field_name"])
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "true": true,
+        "zero": 0,
+        "non_exhaustive": "#[non_exhaustive]",
+    },
+    error_message: "A pub enum's #[non_exhaustive] struct variant has a new field. Pattern matches that explicitly list fields will now fail to compile.",
```

Comment:
> Pattern matches that explicitly list fields will now fail to compile.

Are you sure this is true? Did you test this out? Recall that the struct variant is `#[non_exhaustive]`.

It is extremely important that all these descriptions and messages are 100% correct. I cannot stress this enough! It's 100x better to not have a lint at all, than to have a lint that says the wrong thing to the user.